src/utility/hexgrid.js

```diff
@@ -285,6 +288,50 @@ export class HexGrid {
 		return o;
 	}
 
+	/**
+	 * Return whether there is at least one creature in the hexes that satisfies
+	 * various conditions, e.g. team.
+	 *
+	 * @param {} dir ?
+	 * @param {Object} o
+	 * @return {boolean} At least one valid target.
+	 */
+	atLeastOneTarget(dir, o) {
```

Comment:
There is a fair amount of overlap between this method, `this.getDirectionChoices` and some methods in the `Ability` class. I don't feel I understand the small differences enough to refactor and consolidate at this point, but hopefully soon.